universal-login-react/src/ui/Onboarding/Onboarding.tsx

```diff
@@ -21,6 +22,8 @@ export interface OnboardingProps {
 export const Onboarding = (props: OnboardingProps) => {
   const modalService = useModalService<ReactModalType, ReactModalProps>();
   const [walletService] = useState<WalletService>(props.walletService || new WalletService(props.sdk));
```

Comment:
Why state? Why not planning to change service during the lifecycle of the component?
Simply:
```ts
const walletService = props.walletService || new WalletService(props.sdk);
```

Reply:
That would create a new instance on each re-render, which is most likely not what we want

Reply:
Good point, let's make it required props then.